### ISSUE TITLE
extstore broken build on 32-bit

### ISSUE THREAD

**Issue by remicollet** [memcached/memcached#319]
Tring to build with --enable-extstore option on 32-bit

```
gcc -DHAVE_CONFIG_H -I.    -fprofile-arcs -ftest-coverage -DMEMCACHED_DEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m32 -march=i686 -fasynchronous-unwind-tables -pie -fpie -pthread -pthread -Wall  -pedantic -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -c -o memcached_debug-memcached.o `test -f 'memcached.c' || echo './'`memcached.c
crc32c.c: In function 'crc32c_hw':
crc32c.c:314:1: warning: unsupported size for integer register
 }
 ^
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c:314:1: warning: unsupported size for integer register
crc32c.c: Assembler messages:
crc32c.c:259: Error: unsupported instruction `crc32'
crc32c.c:260: Error: unsupported instruction `crc32'
crc32c.c:261: Error: unsupported instruction `crc32'
crc32c.c:279: Error: unsupported instruction `crc32'
crc32c.c:280: Error: unsupported instruction `crc32'
crc32c.c:281: Error: unsupported instruction `crc32'
crc32c.c:296: Error: unsupported instruction `crc32'
```

**Comment by dormando** [comment]
thanks! that's breaking ARM too. need to gate that file on some compilation tests.

**Comment by sergey-dryabzhinsky** [comment]
The source of error - strange check for availability of SSE4.2 instructions to use HW-accelerated crc32 function.
I suggest to use 'common' crc32 code w/o HW-acceleration.

Simple patch:
```
Index: memcached-1.5.7/crc32c.c
===================================================================
--- memcached-1.5.7.orig/crc32c.c       2018-03-30 08:50:15.419446203 +0300
+++ memcached-1.5.7/crc32c.c    2018-03-30 08:51:50.722490168 +0300
@@ -334,7 +334,11 @@
 void crc32c_init(void) {
     int sse42;
 
+#ifdef USE_SW_CRC32
+    sse42 = 0;
+#else
     SSE42(sse42);
+#endif
     if (sse42) {
         crc32c = crc32c_hw;
     } else {
```
Then compile memcached with `export CFLAGS = -DUSE_SW_CRC32`

**Comment by dormando** [comment]
it's easy enough to do this properly via the configure scripts, I just haven't had time to circle back to it yet.

**Comment by sergey-dryabzhinsky** [comment]
memcached compiles now, but extstore tests fails with:
```
memcached-debug: memcached.c:3625: _get_extstore_cb: Assertion `read_it->slabs_clsid != 0' failed.
```

**Comment by dormando** [comment]
the fixes are released in 1.5.8, thanks!
